examples/tensorflow_simple.py

```diff
@@ -0,0 +1,148 @@
+"""
+Optuna example that optimizes multi-layer perceptrons using Tensorflow.
+
+In this example, we optimize the validation accuracy of hand-written digit recognition using
+Tensorflow and MNIST. We optimize the neural network architecture as well as the optimizer
+configuration. As it is too time consuming to use the whole MNIST dataset, we here use a small
+subset of it.
+
+We have the following two ways to execute this example:
+
+(1) Execute this code directly.
+    $ python tensorflow_simple.py
+
+
+(2) Execute through CLI.
+    $ STUDY_NAME=`optuna create-study --storage sqlite:///example.db`
+    $ optuna study optimize tensorflow_simple.py objective --n-trials=100 --study $STUDY_NAME \
+      --storage sqlite:///example.db
+
+"""
+
+from __future__ import division
+from __future__ import print_function
+
+import numpy as np
+import shutil
+import tensorflow as tf
+
+import optuna
+
+MODEL_DIR = "/tmp/mnist_convnet_model"
+BATCH_SIZE = 128
+TRAIN_STEPS = 1000
```

Comment:
In my local environment, a study takes about 10 minutes, and it is about five times longer than the execution time of Chainer example. If we merge this PR, the execution time of CircleCI's example-pythonXX will be tripled.
Could you tune the parameters like training steps, batch size, and n_trials to reduce the execution time?

Reply:
Okay. I will change `n_trials` to `25`. This value is the same as ChainerMN examples and seems sufficient to find out nearly optimal parameters for Tensorflow examples.

FYI
----

The followings are new execution times of examples on my Windows(WSL) laptop.
```console
$ time -p python3 examples/tensorflow_simple.py
Number of finished trials:  25

real 82.64
user 133.15
sys 62.09

$ time -p python3 examples/chainer_simple.py
Number of finished trials:  100

real 227.17
user 1006.57
sys 1062.59

$ time -p python3 examples/pruning/tensorflow_integration.py
  Number of finished trials:  25
  Number of pruned trials:  13
  Number of complete trials:  12

real 124.54
user 159.37
sys 48.15

$ time -p python3 examples/pruning/chainer_integration.py
Study statistics:
  Number of finished trials:  100
  Number of pruned trials:  90
  Number of complete trials:  10

real 128.71
user 495.78
sys 509.57
```